mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java

```diff
@@ -82,9 +84,7 @@ public void destroy() {
      * @throws NullPointerException before {@link #init(MailetConfig)}
      */
     public boolean getInitParameter(String name, boolean defaultValue) {
-        if (config == null) {
-            throw new NullPointerException("Mailet configuration must be set before getInitParameter is called.");
-        }
+        Preconditions.checkNotNull(config, CONFIG_IS_NULL_ERROR_MESSAGE);
```

Comment:
technically speaking, it should not be a null pointer exception but an illegal state exception